### ISSUE TITLE
amp-subscriptions: change fallback protocol

### ISSUE THREAD

**Issue by dvoytenko** [ampproject/amphtml#18319]
Fallback, as before, should be used for local service. The change, however:

1. Instead of requiring _all_ services to fail - it's enough for local service to fail.
2. However, before we resolve local service to fallback, we should wait for other services to respond.

E.g.

```
onFail(serviceId) {
  if (serviceId == 'local' && hasFallback) {
    waitForAllToRespondSuccessOrFailure().then(() => {
      resolveFailure('local', fallback);
    })
  } else {
    resolveFailure(serviceId, ...);
  }
}
```

**Comment by prateekbh** [comment]
`it's enough for local service to fail` always?

I think if this is the case the `fallbackEntitlement` should be a part of local service config in 
`<script id="amp-subscriptions" type="application/json">....` instead of a global fallback

WDYT?

**Comment by dvoytenko** [comment]
I think this is really what `fallback` really means to me: a page author cannot really do much about other services failing. So `fallback` really applies to local. I think we had it in the global level only b/c it looked "flatter". Looking at the code - it definitely currently applies as local. At this point, I'd rather fix/address, but not yet move where it is in config.

**Comment by prateekbh** [comment]
Cool makes sense

**Comment by prateekbh** [comment]
Here's a problem though:
By design we only give a platforms config to the platform. Anything outside will require some change in design.

**Comment by dvoytenko** [comment]
I think it's fine to keep `fallback` logic in the platform-store - this is where we assign it to local anyway. We just need to change sequence a little to avoid race condition.

**Comment by prateekbh** [comment]
>Instead of requiring all services to fail 

if the fallback entitlement says granted:true then by our rule "subscriber always and immediately wins", the fallback entitlement wins immediately.

Even if there are more services saying that the user is s subscriber.

That'd also be why we waited for every one to fail.

**Comment by prateekbh** [comment]
should be fixed by #18363
